editor/effects.js

```diff
@@ -27,17 +28,30 @@ export default {
 			...edits,
 			content: serialize( getBlocks( state ) ),
 		};
+		const transactionId = uniqueId();
 
 		if ( ! isNew ) {
 			toSend.id = postId;
 		}
 
-		dispatch( { type: 'CLEAR_POST_EDITS' } );
+		dispatch( {
+			type: 'CLEAR_POST_EDITS',
+			optimist: { type: BEGIN, id: transactionId },
+		} );
+		dispatch( {
+			type: 'UPDATE_POST',
+			edits: toSend,
+			optimist: { id: transactionId },
+		} );
 		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
 			dispatch( {
 				type: 'REQUEST_POST_UPDATE_SUCCESS',
-				post: newPost,
```

Comment:
Removing this is causing an error when save completes:

![image](https://user-images.githubusercontent.com/1779930/27037119-f943afe0-4f54-11e7-89f9-8e2459940159.png)